src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -93,18 +94,70 @@ def export_group_as_template(
 
 def deploy_arm_template(
         resource_group_name, template_file=None, template_uri=None, deployment_name=None,
-        parameters=None, mode='incremental', no_wait=False):
+        parameters=None, mode='incremental', no_wait=False, prompt_for_parameters=True):
```

Comment:
With this, prompting is on by default.
In https://github.com/Azure/azure-cli/issues/1778#issuecomment-279834290, looks like the request was to only prompt if `--prompt` is supplied.